gtk4/src/subclass/widget.rs

```diff
@@ -895,6 +895,16 @@ pub unsafe trait WidgetClassSubclassExt: ClassStruct {
         }
     }
 
+    fn add_shortcut(&mut self, shortcut: &Shortcut) {
+        unsafe {
+            let type_class = self as *mut _ as *mut glib::gobject_ffi::GTypeClass;
+            let widget_class =
+                glib::gobject_ffi::g_type_check_class_cast(type_class, ffi::gtk_widget_get_type())
+                    as *mut ffi::GtkWidgetClass;
```

Comment:
Now my remaining question: why so complicated? :) We already statically ensured that the type is correct. You can simply cast `self as [...] as *mut ffi::GtkWidgetClass` instead of doing the runtime type check.

Reply:
I guess that went unnoticed from the composite template PR :)